bindings_test.go

```diff
@@ -930,3 +930,137 @@ func TestFunctionParameters(t *testing.T) {
 		}
 	})
 }
+
+func TestVariousBindingModesForNonNullValues(t *testing.T) {
+	testcases := []struct {
+		testDesc  string
+		paramType string
+		input     interface{}
+	}{
+		{"textAndString", "text", "string"},
+		{"numberAndInteger", "number", 123},
+		{"floatAndFloat", "float", 123.01},
+		{"booleanAndBoolean", "boolean", true},
+		//{"dateAndTime", "date", time.Now()}, // does not bind in any binding mode
+		{"datetimeAndTime", "datetime", time.Now()},
+		//{"timeAndTime", "time", time.Now()}, // does not bind in any binding mode
+		{"timestampAndTime", "timestamp", time.Now()},
+		{"timestamp_ntzAndTime", "timestamp_ntz", time.Now()},
+		{"timestamp_ltzAndTime", "timestamp_ltz", time.Now()},
+		{"timestamp_tzAndTime", "timestamp_tz", time.Now()},
+	}
+
+	bindingModes := []struct {
+		desc      string
+		insert    string
+		query     string
+		transform func(interface{}) interface{}
+	}{
+		{
+			desc:      "?",
+			insert:    "INSERT INTO BINDING_MODES VALUES(?)",
+			query:     "SELECT * FROM BINDING_MODES WHERE param1 = ?",
+			transform: func(v interface{}) interface{} { return v },
+		},
+		{
+			desc:      ":1",
+			insert:    "INSERT INTO BINDING_MODES VALUES(:1)",
+			query:     "SELECT * FROM BINDING_MODES WHERE param1 = :1",
+			transform: func(v interface{}) interface{} { return v },
+		},
+		{
+			desc:      ":id",
+			insert:    "INSERT INTO BINDING_MODES VALUES(:param)",
+			query:     "SELECT * FROM BINDING_MODES WHERE param1 = :param",
+			transform: func(v interface{}) interface{} { return sql.Named("param", v) },
+		},
+	}
+
+	runTests(t, dsn, func(dbt *DBTest) {
+		for _, tc := range testcases {
+			for _, bindingMode := range bindingModes {
+				t.Run(tc.testDesc+" "+bindingMode.desc, func(t *testing.T) {
+					query := fmt.Sprintf(`CREATE OR REPLACE TABLE BINDING_MODES(param1 %v)`, tc.paramType)
+					dbt.mustExec(query)
+					if _, err := dbt.db.Exec(bindingMode.insert, bindingMode.transform(tc.input)); err != nil {
+						t.Fatal(err)
+					}
+					rows, err := dbt.db.Query(bindingMode.query, bindingMode.transform(tc.input))
+					if err != nil {
+						t.Fatal(err)
+					}
+					if !rows.Next() {
+						t.Fatal("Expected to return a row")
+					}
+				})
+			}
+		}
+	})
+}
+
+func TestVariousBindingModesForNullValues(t *testing.T) {
```

Comment:
what's the difference between these suite and the one above?

Reply:
There are nulls tested. Nulls are handled in a different manner in Go which supports nulls only for pointers. Also, the query is also different.

Reply:
In the new test there is a lot of duplication - maybe you can extract one parameter more and then put those tests in one suite?

Reply:
I tried - but it's less readable. You have to change query between `= ?`/`= :1`/`= :1` and `is null`. IMHO it wasn't worth it.

Reply:
I think I would consolidate these. Seems like theres no point in maintaining the same test running logic twice. Also It seems like we would only add some scenarios' config so readability wouldn't suffer that much